core/tracepoint/disable_spec.rb

```diff
@@ -0,0 +1,57 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.0' do
+  describe 'TracePoint#disable' do
+    def test; end
+    it 'returns true if trace was enabled' do
+      event_name, method_name = nil
+      trace = TracePoint.new(:call) do |tp|
+        event_name = tp.event
+        method_name = tp.method_id
+      end
+
+      trace.enable
+      trace.disable.should be_true
+      event_name, method_name = nil
+      test
+      method_name.equal?(:test).should be_false
+      event_name.should equal(nil)
+    end
+
+    it 'returns false if trace was disabled' do
+      event_name, method_name = nil
+      trace = TracePoint.new(:call) do |tp|
+        event_name = tp.event
+        method_name = tp.method_id
+      end
+
+      trace.disable.should be_false
+      event_name, method_name = nil
+      test
+      method_name.equal?(:test).should be_false
+      event_name.should equal(nil)
+    end
+
+    it 'is disabled within a block' do
+      event_name = nil
+      trace = TracePoint.new(:line) {}
+      trace.enable
+      trace.disable { puts 2 + 2 }.should == nil
```

Comment:
Debug `puts` left in here and also in `enable` specs.
Could you instead check if the trace is disabled/enabled in the block? Like so:
```ruby
enabled = nil
trace.disable { enabled = trace.enabled? }
enabled.should == false
```

Reply:
How about this ?--
For TracePoint#disable
```
trace.disable { trace.enabled? }.should == false
```
For TracePoint#enable
```
trace.enable { trace.enabled? }.should == true
```
I ran the specs & they pass.

Reply:
Ah, enable/disable return the value of the block?
It's still a bit risky as this spec could accidentally work if `enable/disable` just always returned `true/false`.

It would be interesting to test the return value of `enable/disable` with a block in another spec example though :)

Reply:
okay, I have updated the spec & add another spec example to check the return value of enable/disable with a block. I have updated the PR.